src/parser/ParsedQuery.h

```diff
@@ -360,28 +343,25 @@ class ParsedQuery {
       return std::get<std::vector<string>>(_varsOrAsterisk);
     }
 
-    void pushVariablesOrder(const string& variable) {
+    // Add a variable, that was found in the query body. The added variables
+    // will only be used if `isAsterisk` is true.
+    void addVariableFromQueryBody (const string& variable) {
       if(!(std::find(_variablesOrder.begin(),
                     _variablesOrder.end(),
                     variable) != _variablesOrder.end())) {
         _variablesOrder.emplace_back(variable);
       }
     }
 
-
-    [[nodiscard]] std::list<string>& retrieveOrder() {
-      return _variablesOrder;
-    }
-
-    [[nodiscard]] std::list<string> retrieveOrder() const {
+    [[nodiscard]] auto retrieveOrder() const {
```

Comment:
Here one of my comments disappeared:
Call this method `orderedVariablesFromQueryBody` and comment it:
`// get the variables which `addVariableFromQueryBody` was previously called. The result contains no duplicates and is ordered by the first appearance in the query body.